### ISSUE TITLE
[RyuJIT/ARM32] Wrong generated code when processing PutArgSplit

### ISSUE THREAD

**Issue by YongseopKim** [dotnet/coreclr#12703]
cs
```cs
using System;

namespace regs_stack
{
    public struct MyStruct
    {
        public int a;
        public int b;
        public int c;
        public int d;
    }

    public struct MyStruct2
    {
        public long a;
        public long b;
    }

    class Program
    {
        static public void CallMyStruct(int i, MyStruct st)
        {
            if (st.a != 10 || st.b != 20 || st.c != 30 || st.d != 40)
            {
                System.Console.WriteLine("Failed in CallMyStruct");
            }
        }

        static public void CallMyStruct2(int i, MyStruct2 st2)
        {
            if (st2.a != 100L || st2.b != 1000L)
            {
                System.Console.WriteLine("Failed in CallMyStruct2");
            }
        }

        static void Main()
        {
            int i = 55;

            MyStruct st;
            st.a = 10;
            st.b = 20;
            st.c = 30;
            st.d = 40;
            CallMyStruct(i, st);

            MyStruct2 st2;
            st2.a = 100L;
            st2.b = 1000L;
            CallMyStruct2(i, st2);
        }
    }
}
```

Run
```
Failed in CallMyStruct
Failed in CallMyStruct2
```

Jitdump of CallMyStruct
```asm
*************** In lvaAssignFrameOffsets(FINAL_FRAME_LAYOUT)
Assign V00 loc0, size=4, stkOffs=-0x14
Assign V01 loc1, size=16, stkOffs=-0x24
Assign V02 loc2, size=16, stkOffs=-0x38
Assign V04 tmp1, size=4, stkOffs=-0x3c
Assign V05 tmp2, size=4, stkOffs=-0x40
Assign V07 tmp4, size=4, stkOffs=-0x44
Assign V06 tmp3, size=4, stkOffs=-0x48
Assign V03 OutArgs, size=8, stkOffs=-0x50
; Final local variable assignments
;
;  V00 loc0         [V00,T06] (  3,  3   )     int  ->  [sp+0x3C]   do-not-enreg[] must-init
;  V01 loc1         [V01,T00] (  8,  8   )  struct (16) [sp+0x2C]   do-not-enreg[SFA] multireg-arg must-init ld-addr-op
;  V02 loc2         [V02,T01] (  7,  7   )  struct (16) [sp+0x18]   do-not-enreg[SFA] multireg-arg must-init ld-addr-op double-align
;  V03 OutArgs      [V03    ] (  1,  1   )  lclBlk ( 8) [sp+0x00]   do-not-enreg[]
;  V04 tmp1         [V04,T03] (  3,  6   )     int  ->  [sp+0x14]   do-not-enreg[] must-init
;  V05 tmp2         [V05,T04] (  3,  6   )     int  ->  [sp+0x10]   do-not-enreg[] must-init
;  V06 tmp3         [V06,T02] (  3,  6   )   byref  ->  [sp+0x08]   do-not-enreg[] must-init
;  V07 tmp4         [V07,T05] (  2,  4   )     int  ->  [sp+0x0C]   do-not-enreg[] must-init
;
; Lcl frame size = 64


Added IP mapping: 0x0028 STACK_EMPTY (G_M60777_IG03,ins#14,ofs#28)
Generating: N073 ( 34, 37) [000040] -------------                IL_OFFSET void   IL offset: 0x28 REG NA
Generating: N075 (  3,  4) [000088] -------------       t88 =    LCL_FLD   int    V01 loc1         [+0] r3 REG r3
IN0017:             ldr     r3, [sp+0x2c]   // [V01 loc1]
Generating: N077 (  3,  4) [000090] -------------       t90 =    LCL_FLD   int    V01 loc1         [+4] r2 REG r2
IN0018:             ldr     r2, [sp+0x30]   // [V01 loc1+0x04]
Generating: N079 (  3,  4) [000092] -------------       t92 =    LCL_FLD   int    V01 loc1         [+8] r1 REG r1
IN0019:             ldr     r1, [sp+0x34]   // [V01 loc1+0x08]
Generating: N081 (  3,  4) [000094] -------------       t94 =    LCL_FLD   int    V01 loc1         [+12] r0 REG r0
IN001a:             ldr     r0, [sp+0x38]   // [V01 loc1+0x0c]
                                                              /--*  t88    int
                                                              +--*  t90    int
                                                              +--*  t92    int
                                                              +--*  t94    int
Generating: N083 ( 15, 23) [000089] -c-----------       t89 = *  FIELD_LIST void   int at offset 0 REG NA
                                                              /--*  t89    void
Generating: N085 (???,???) [000145] -------------      t145 = *  PUTARG_SPLIT struct REG r1
IN001b:             mov     r1, r3                                    ; This is weird
IN001c:             mov     r3, r1                                    ; too
IN001d:             str     r0, [sp]    // [V03 OutArgs]
Generating: N087 (  3,  2) [000033] -------------       t33 =    LCL_VAR   int    V00 loc0          r0 REG r0
IN001e:             ldr     r0, [sp+0x3c]   // [V00 loc0]
                                                              /--*  t33    int
Generating: N089 (???,???) [000146] -------------      t146 = *  PUTARG_REG int    REG r0
Generating: N091 (  3,  7) [000147] -------------      t147 =    CNS_INT(h) int    0x744F55C8 ftn REG lr
IN001f:             movw    lr, 0x55c8
IN0020:             movt    lr, 0x744f
                                                              /--*  t147   int
Generating: N093 (  6,  9) [000148] -------------      t148 = *  IND       int    REG lr
IN0021:             ldr     lr, [lr]
                                                              /--*  t145   struct arg1 r1-r3 out+00
                                                              +--*  t146   int    arg0 in r0
                                                              +--*  t148   int    control expr
Generating: N095 ( 34, 37) [000035] --C-G--------             *  CALL      void   regs_stack.Program.CallMyStruct
Call: GCvars=00000000 {}, gcrefRegs=0000 {}, byrefRegs=0000 {}
Added IP mapping: 0x002A CALL_INSTRUCTION (G_M60777_IG03,ins#25,ofs#56)
IN0022:             blx     lr      // regs_stack.Program:CallMyStruct(int,struct)
```

```asm
*************** In lvaAssignFrameOffsets(FINAL_FRAME_LAYOUT)
Assign V00 loc0, size=4, stkOffs=-0x14
Assign V01 loc1, size=16, stkOffs=-0x24
Assign V02 loc2, size=16, stkOffs=-0x38
Assign V04 tmp1, size=4, stkOffs=-0x3c
Assign V05 tmp2, size=4, stkOffs=-0x40
Assign V07 tmp4, size=4, stkOffs=-0x44
Assign V06 tmp3, size=4, stkOffs=-0x48
Assign V03 OutArgs, size=8, stkOffs=-0x50
; Final local variable assignments
;
;  V00 loc0         [V00,T06] (  3,  3   )     int  ->  [sp+0x3C]   do-not-enreg[] must-init
;  V01 loc1         [V01,T00] (  8,  8   )  struct (16) [sp+0x2C]   do-not-enreg[SFA] multireg-arg must-init ld-addr-op
;  V02 loc2         [V02,T01] (  7,  7   )  struct (16) [sp+0x18]   do-not-enreg[SFA] multireg-arg must-init ld-addr-op double-align
;  V03 OutArgs      [V03    ] (  1,  1   )  lclBlk ( 8) [sp+0x00]   do-not-enreg[]
;  V04 tmp1         [V04,T03] (  3,  6   )     int  ->  [sp+0x14]   do-not-enreg[] must-init
;  V05 tmp2         [V05,T04] (  3,  6   )     int  ->  [sp+0x10]   do-not-enreg[] must-init
;  V06 tmp3         [V06,T02] (  3,  6   )   byref  ->  [sp+0x08]   do-not-enreg[] must-init
;  V07 tmp4         [V07,T05] (  2,  4   )     int  ->  [sp+0x0C]   do-not-enreg[] must-init
;
; Lcl frame size = 64

Added IP mapping: 0x0047 STACK_EMPTY (G_M60777_IG03,ins#48,ofs#104)
Generating: N155 ( 34, 37) [000064] -------------                IL_OFFSET void   IL offset: 0x47 REG NA
Generating: N157 (  3,  4) [000102] -------------      t102 =    LCL_FLD   int    V02 loc2         [+0] r3 REG r3
IN0039:             ldr     r3, [sp+0x18]   // [V02 loc2]
Generating: N159 (  3,  4) [000104] -------------      t104 =    LCL_FLD   int    V02 loc2         [+4] r2 REG r2
IN003a:             ldr     r2, [sp+0x1c]   // [V02 loc2+0x04]
Generating: N161 (  3,  4) [000106] -------------      t106 =    LCL_FLD   int    V02 loc2         [+8] r0 REG r0
IN003b:             ldr     r0, [sp+0x20]   // [V02 loc2+0x08]
Generating: N163 (  3,  4) [000108] -------------      t108 =    LCL_FLD   int    V02 loc2         [+12] r1 REG r1
IN003c:             ldr     r1, [sp+0x24]   // [V02 loc2+0x0c]
                                                              /--*  t102   int
                                                              +--*  t104   int
                                                              +--*  t106   int
                                                              +--*  t108   int
Generating: N165 ( 15, 23) [000103] -c-----------      t103 = *  FIELD_LIST void   int at offset 0 REG NA
                                                              /--*  t103   void
Generating: N167 (???,???) [000149] -------------      t149 = *  PUTARG_SPLIT struct REG r2
IN003d:             mov     r2, r3
IN003e:             mov     r3, r2
IN003f:             str     r0, [sp]    // [V03 OutArgs]
IN0040:             str     r1, [sp+0x04]   // [V03 OutArgs+0x04]
Generating: N169 (  3,  2) [000057] -------------       t57 =    LCL_VAR   int    V00 loc0          r0 REG r0
IN0041:             ldr     r0, [sp+0x3c]   // [V00 loc0]
                                                              /--*  t57    int
Generating: N171 (???,???) [000150] -------------      t150 = *  PUTARG_REG int    REG r0
Generating: N173 (  3,  7) [000151] -------------      t151 =    CNS_INT(h) int    0x744F55E8 ftn REG r1
IN0042:             movw    r1, 0x55e8
IN0043:             movt    r1, 0x744f
                                                              /--*  t151   int
Generating: N175 (  6,  9) [000152] -------------      t152 = *  IND       int    REG r1
IN0044:             ldr     r1, [r1]
                                                              /--*  t149   struct arg1 r2,r3 out+00
                                                              +--*  t150   int    arg0 in r0
                                                              +--*  t152   int    control expr
Generating: N177 ( 34, 37) [000059] --C-G--------             *  CALL      void   regs_stack.Program.CallMyStruct2
Call: GCvars=00000000 {}, gcrefRegs=0000 {}, byrefRegs=0000 {}
```

**Comment by YongseopKim** [comment]
@BruceForstall , I know I'm not included in dotnet/arm32-contrib. Could you add me, @wateret and @jyoungyun into the dotnet/arm32-contrib, please?

**Comment by BruceForstall** [comment]
> @BruceForstall , I know I'm not included in dotnet/arm32-contrib. Could you add me, @wateret and @jyoungyun into the dotnet/arm32-contrib, please?

Done. FYI, you can see all the members here: https://github.com/orgs/dotnet/teams/arm32-contrib/members

**Comment by YongseopKim** [comment]
@BruceForstall Thank you so much!

**Comment by hseok-oh** [comment]
@YongseopKim Fixed by #12710. Please close this issue.
